src/wallet.rs

```diff
@@ -0,0 +1,424 @@
+
+        // Create a descriptor map for the presentation submission based on the credentials
+        // returned from the VDC collection.
+        //
+        // NOTE: The order of the descriptor map is important as the index of the descriptor
+        // in the presentation submission must match the index of the credential in the
+        // presentation submission.
+        //
+        // For example, when adding to the `verifiableCredentials` array in the presentation
+        // submission, the order of the credentials must match the order of the descriptors
+        // in the descriptor map where there paths are index-based.
+        let credential_descriptor_map = self.create_descriptor_maps(credentials);
+
+        // Create a presentation submission.
+        let presentation_submission = PresentationSubmission::new(
+            presentation_submission_id,
+            presentation_definition_id,
+            // Use the descriptor map to create the submission.
+            credential_descriptor_map
+                .iter()
+                .map(|(descriptor, _)| descriptor.clone())
+                .collect(),
+        )
+        .try_into()
+        .map_err(|e: anyhow::Error| WalletError::PresentationSubmissionCreation(e.to_string()))?;
+
+        let vp_token = self
+            .create_verifiable_presentation_jwt(credential_descriptor_map, key_index)
+            .await?;
+
+        // Create a verifiable presentation object.
+        Ok(UnencodedAuthorizationResponse(
+            Default::default(),
+            VpToken(vp_token),
+            presentation_submission,
+        ))
+    }
+
+    // Internation method for creating a verifiable presentation JWT.
+    async fn create_verifiable_presentation_jwt(
+        &self,
+        credential_descriptor_map: Vec<(DescriptorMap, Credential)>,
+        key_index: Option<u8>,
+    ) -> Result<String, WalletError> {
+        let verifiable_credential = credential_descriptor_map
+            .into_iter()
+            .map(|(_, credential)| credential)
+            .collect::<Vec<Credential>>();
+
+        let vp_json = json!({
+            "@context": [
+                "https://www.w3.org/2018/credentials/v1",
+                "https://www.w3.org/2018/credentials/examples/v1"
+            ],
+            "type": [
+                "VerifiablePresentation"
+            ],
+            "verifiableCredential": verifiable_credential
+        });
+
+        let vp: Presentation = serde_json::from_value(vp_json)
+            .map_err(|e| WalletError::VerifiablePresentation(e.to_string()))?;
+
+        let ldp_options = self.new_linked_data_proof_options(key_index)?;
+
+        // Constuct unsigned JWT, then sign it with the key manager.
+        let unsigned_jwt_vp = vp
+            .generate_jwt(None, &ldp_options, did_jwk::DIDJWK.to_resolver())
+            .await
+            .map_err(|e| WalletError::GenerateJwt(e.to_string()))?;
+
+        unimplemented!()
```

Comment:
NOTE: working through the latest ssi-* crate API changes for generating unsigned jwt_vp. Taking inspiration from companion test wallet, but using the latest ssi-* crates. There's a few API changes in the latest versions, versus the flow used in the companion wallet based on the prior releases.

Reply:
After fighting a bit with the versioning, I decided to use the latest `JWTClaims` API for signing.

Working on testing. This is using the latest ssi-* versions and verifiable presentation creation flow based on the oid4vp e2e test.